lib/pages/timetable/view/events/add_event_view.dart

```diff
@@ -1,765 +1,792 @@
+    Provider.of<UniEventProvider>(context, listen: false)
+        .fetchCalendars()
+        .then((calendars) {
+      setState(() {
+        this.calendars = calendars;
+        selectedCalendar = calendars.keys.first;
+      });
+
+      if (widget.initialEvent?.id != null) {
+        selectedCalendar = widget.initialEvent.calendar.id;
+        final AllDayUniEvent secondSemester =
+            widget.initialEvent.calendar.semesters.last;
+        selectedSemester =
+            Interval(secondSemester.startDate, secondSemester.endDate)
+                    .includes(widget.initialEvent.start)
+                ? 2
+                : 1;
+      } else {
+        bool foundSemester = false;
+        for (final calendar in calendars.entries) {
+          for (final semester in calendar.value.semesters) {
+            final DateTime date = widget.initialEvent.start ?? DateTime.now();
+            if (date.isBeforeOrDuring(semester)) {
+              // semester.id is represented as "semesterN", where "semester0" is the first semester
+              selectedSemester =
+                  1 + int.tryParse(semester.id[semester.id.length - 1]);
+              selectedCalendar = calendar.key;
+              foundSemester = true;
+              break;
+            }
+          }
+          if (foundSemester) break;
+        }
+        if (!foundSemester) {
+          selectedCalendar = calendars.entries.last.value.id;
+          selectedSemester = 2;
+        }
+      }
+
+      if (widget.initialEvent != null &&
+          widget.initialEvent is RecurringUniEvent) {
+        final RecurringUniEvent event = widget.initialEvent;
+        if (event.rrule.interval != 1) {
+          final rule = WeekYearRules.iso;
+          if (rule.getWeekOfWeekYear(LocalDate.dateTime(semester.start)) ==
+              rule.getWeekOfWeekYear(LocalDate.dateTime(event.start))) {
+            // Week is odd
+            weekSelected[WeekType.even] = false;
+            weekSelected[WeekType.odd] = true;
+          } else {
+            // Week is even
+            weekSelected[WeekType.even] = true;
+            weekSelected[WeekType.odd] = false;
+          }
+        }
+      }
+
+      setState(() {
+        weekSelected[WeekType.even] ??= true;
+        weekSelected[WeekType.odd] ??= true;
+      });
+    });
+
+    selectedEventType = widget.initialEvent?.type;
+    selectedClass = widget.initialEvent?.classHeader;
+    selectedTeacher = widget.initialEvent is ClassEvent
+        ? (widget.initialEvent as ClassEvent).teacher
+        : null;
+    locationController =
+        TextEditingController(text: widget.initialEvent?.location ?? '');
+
+    final startHour = widget.initialEvent?.start?.hour ?? 8;
+    duration = widget.initialEvent?.period?.toTime()?.toDuration ??
+        const Duration(hours: 2);
+    startDateTime = widget.initialEvent?.start
+            ?.copyWith(hour: startHour, minute: 0, second: 0, millisecond: 0)
+            ?.copyWithUtc() ??
```

Comment:
This shouldn't be necessary if we do it in the provider (see my corresponding comment). Ideally we want to do this in one place, cause if we rely on people dealing with start/end dates to remember to do it, we're bound to create bugs.